OptimoveSDK/Sources/Classes/Optimobile/Optimobile+Push.swift

```diff
@@ -209,27 +209,22 @@ extension Optimobile {
         Parameters:
             - notification: The notification which triggered the action
     */
-    static func pushTrackOpen(notification: PushNotification?) {
-        guard let notification = notification else {
+    static func pushTrackOpen(notification: PushNotification) {
+        if notification.id == 0 {
+            Logger.error("""
```

Comment:
Is this always an error?

For example, we may be passed a notification from another push provider, and the ID is (I think) expected to be 0. (unless the reason for optionals was if the initialiser of the PushNotification model is optional?)

Reply:
Currently, the initialization of the PushNotification model returns a non-optional object with default values, where the `id` is set to 0. However, the check `notification.id == 0` is not accurate for checking Optimove payload. To improve this, we should apply the `Swift.Decodable` protocol and throw the corresponding error as a sign of validation. 

For now, I can leave it as it is or apply `Swift.Decodable` to `PushNotification`. Either way, the error handling will remain the same.